doc/life_of_a_language_feature.md

```diff
@@ -230,26 +262,64 @@ switched on.
 Details of how to do this will be described in a separate (Google internal)
 document.
 
-## Shipping
+### Implementation Work
+
+This is the bulk of creating a new language feature and this stage will take the
+most time.
+
+Here are some tips for the implementation stage:
 
-Implemented features will be released according to the implementation plan.
-The language team will contribute to:
+- All implementation works should be under the experiment flag.
+- Use the language tests as a pseudo-checklist for what's left to
+implement.
+- Some teams may have their own checklist for what needs to be considered in a
+new language feature implementation. For example, 
+[this analyzer doc](https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/doc/process/new_language_feature.md) 
+and [this analysis server doc](https://github.com/dart-lang/sdk/blob/main/pkg/analysis_server/doc/process/new_language_feature.md).
+- As we work through an implementation, we may encounter problems or holes in
+the spec. If so, create a new issue, add the feature label to it, and cc
+@dart-lang/language-team to discuss if we should make any changes in the
+specification.
 
-  - Helping internal and external teams through any required migration.
+> [!NOTE]
+> Remember that making a language feature is a two way street between the spec
+> writers and the implementers. The implementation may change because the spec
+> changed, or the spec might change after finding out the implementation is too
+> complex.
 
-  - Communicating and advertising the change.
+## Migrations
 
-  - Documenting the change.
+Once the implementation is finished, you should start a CL that enables the
+feature by default, if you haven't already. Use this CL to check what code your
+feature will break and determine if you need any other lints or quick-fixes to
+help these migrations.
 
-  - Releasing the change.
+See [Google3 testing](#google3-testing) for more information on fixing breakages
+within Google3.
+
+## Shipping
 
-## After shipping
+This is a general checklist for shipping a feature:
+- [ ] Make sure all pre-migrations are complete.
+- [ ] Submit a CL that enables the feature by default in the upcoming version.
+- [ ] Follow up on documentation changes which should come out when the stable
+version with your feature comes out.
+- [ ] Communicate with the team that there's a new feature and invite everyone
+to test it out.
+
+Congratulations! You've now shipped the feature.
+
+Keep an eye out for breakages and any new QOL requests that surface from the
+usage of the feature. Sometimes people use features in ways we don't expect and
+we can learn a lot from seeing how the feature is used in the real world.
+
+### After shipping
 
 After a feature has been shipped, the documents pertaining to the feature should
 be moved into subfolders carrying the name of the release in which they were
 shipped (e.g. `2.1`).
 
-## Sample file layout
+#### Sample file layout
```

Comment:
Nit: I think `##` is better here since this heading isn't subordinate to `### After shipping`.

Or, if you want to be separate but not as emphasized as `##` gives you, you could maybe do:

```md
## Appendix

### Example file layout

...
```

But that feels a bit like overkill to me since there's nothing else in there.

Reply:
Hmm, its more of an example of what's mentioned in `After shipping`. I'll just change the heading to a sentence that's more like "here's an example:".